neo/rawio/tdtrawio.py

```diff
@@ -181,15 +209,24 @@ def _parse_header(self):
                         assert self._sigs_lengths[seg_index][stream_index] == size
 
                     # signal start time, relative to start of segment
-                    t_start = data_index['timestamp'][0]
+                    if len(data_index['timestamp']):
+                        t_start = data_index['timestamp'][0]
+                    else:
+                        # if no signal present use segment t_start as dummy value
+                        t_start = self._seg_t_starts[seg_index]
                     if stream_index not in self._sigs_t_start[seg_index]:
                         self._sigs_t_start[seg_index][stream_index] = t_start
                     else:
                         assert self._sigs_t_start[seg_index][stream_index] == t_start
 
                     # sampling_rate and dtype
-                    _sampling_rate = float(data_index['frequency'][0])
-                    _dtype = data_formats[data_index['dataformat'][0]]
+                    if len(data_index):
+                        _sampling_rate = float(data_index['frequency'][0])
+                        _dtype = data_formats[data_index['dataformat'][0]]
+                    else:
+                        # if no signal present use dummy values
+                        _sampling_rate = 1.
+                        _dtype = int
```

Comment:
Is it safe.

Reply:
These dummy values are used when within a given stream and segment no data were recorded on a channel. In that case no data should be read from file in any case, so the value of the dummy dtype does not matter. It is only relevant for ensuring that all channels in a stream in a segment have the same characteristics (dtype, sampling_rate).